File: Medium/Bit manipulation/3514. Number of Unique XOR Triplets II.py
# 3514. Number of Unique XOR Triplets II
class Solution:
    def uniqueXorTriplets(self, nums: List[int]) -> int:
        st0 = set(nums)
        n = len(st0)
        if(n<3): return n
        st = set()
        for i in st0:
            for j in st0:
                st.add(i ^ j)
        
        st2 = set()

        for i in nums:
            for j in st:
                st2.add(i^j)
        return len(st2)
        

File: Medium/Bit manipulation/3514. Number of Unique XOR Triplets II.cpp
// 3514. Number of Unique XOR Triplets II
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        //return 0;
        
        int n = nums.size();
        unordered_set<int>st(nums.begin(), nums.end());
        
        if(st.size() < 3) return st.size();

        unordered_set<int>st2;

        

        for(auto i: st){
            for(auto j : st){
                if(i == j) continue;
                
                st2.insert(i ^ j);
                

            }
        }
        

        unordered_set<int>st3;

        for(auto i: st){
            for(auto j : st2){
                st3.insert(i^j);
            }
        }
 
 

        return st3.size();
 
    }
};
